app.js

```diff
@@ -1,14 +1,36 @@
 const express = require('express');
 const { getTopics } = require('./controllers/topics-controllers');
+const { getArticleById } = require('./controllers/articles-controllers');
 const { getEndpoints } = require('./controllers/api-controllers');
 const app = express();
 
+
 app.get('/api/topics', getTopics);
 
 app.get('/api', getEndpoints);
 
+app.get('/api/articles/:article_id', getArticleById);
+
+
+app.use((err, request, response, next) => {
+    if (err.code === '22P02') {
+        response.status(400).send({ msg: 'bad request' })
+    } else {
+        next(err);
+    }
+})
+
+app.use((err, request, response, next) => {
+    if (err.status && err.msg) {
+        response.status(err.status).send({ msg: err.msg });
+    } else {
+        next(err);
+    }
+})
+
```

Comment:
I would place your error handling underneath the catch all for invalid endpoints. Whilst it might not make a difference to how it runs, due to that custom error handler being related to endpoints, it should be grouped with the others. 

You also should add in a handleServerErrors error handler to help with debugging future errors